File: backend/src/main/java/com/novatech/backend/service/ProjectService.java
package com.novatech.backend.service;

import com.novatech.backend.dto.ProjectDTO;
import com.novatech.backend.dto.ProjectResponseDTO;
import java.util.List;

import org.springframework.web.multipart.MultipartFile;

public interface ProjectService {

    ProjectResponseDTO createProject(ProjectDTO projectDTO, MultipartFile file, String managerEmail);
    List<ProjectResponseDTO> getAllProjects();

    List<ProjectResponseDTO> getProjectsByManager(Long managerId);
    
    ProjectResponseDTO getProjectById(Long projectId);

    ProjectResponseDTO updateProject(Long projectId, ProjectDTO projectDTO, MultipartFile file, String userEmail);
    void deleteProject(Long projectId, String userEmail);
}

File: frontend/src/services/projectService.js
import api from './api'; 

const createProject = (formData) => {
  return api.post('/projects', formData);
};

const getAllProjects = () => {
  return api.get('/projects');
};

const updateProject = (id, formData) => {
  return api.put(`/projects/${id}`, formData);
};

const deleteProject = (id) => {
  return api.delete(`/projects/${id}`);
};

export default {
  createProject,
  getAllProjects,
  updateProject,
  deleteProject,
};